Modules/_blake2/blake2b_impl.c

```diff
@@ -391,47 +391,24 @@ py_blake2b_dealloc(PyObject *self)
         PyThread_free_lock(obj->lock);
         obj->lock = NULL;
     }
+    PyTypeObject *tp = Py_TYPE(self);
+    Py_DECREF(tp);
```

Comment:
Please move it after PyObject_Del().

Reply:
@vstinner For my own information what's the reason?  Is this because the ref count of self may hit 0 and it will be double freed?

Reply:
Addressed in the blake2 PR